src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs

```diff
@@ -45,6 +46,17 @@ protected override bool OnBackButtonPressed()
             return base.OnBackButtonPressed();
         }
 
+        void ActivateTimePicker(object sender, EventArgs args)
+        {
+            var stackLayout = (ExtendedStackLayout)sender;
+            SettingsPageListItem item = (SettingsPageListItem)stackLayout.BindingContext;
+            if (item.ShowTimeInput)
+            {
+                var timePicker = stackLayout.Children.Where(x => x is TimePicker).FirstOrDefault();
```

Comment:
What if we add more time pickers down the line?

Reply:
I don't think that will happen. The setting page is a bit messy as it is and we should think of ways to re-design it in the long run.

Reply:
Looking forward to that as well! Until then, don't we have an ID or something we could use to pinpoint the exact item here?

Reply:
There might be some confusion here. The `sender` will actually be the `ExtendedStackLayout` inside the `DataTemplate` where the `TimePicker` lives. Its not the `CollectionView` with the list of cells. This means that while there may be additional `TimePicker` cells added, the sender will always refer back to the one tapped. 

We could hardcode the `View` # inside the `ExtendedStackLayout`, but that seems like a bad idea if we ever want to reformat the cell.

Reply:
Misunderstood, thanks for clearing it up!